compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1155,6 +1155,11 @@ object RefChecks {
       then report.warning(ExtensionNullifiedByMember(sym, target.typeSymbol), sym.srcPos)
   end checkExtensionMethods
 
+  def checkSynchronizedCallOnValue(tree: Tree)(using Context): Unit =
+    if tree.symbol == defn.Object_synchronized
+      && ctx.owner.enclosingClass.isValueClass then
```

Comment:
This won't do the right thing for an `import`ed `synchronized`.

You shouldn't look at the context owner, but at the `prefix` of `tree.tpe`.